lib/Logger.php

```diff
@@ -25,17 +26,27 @@ class Logger extends \Psr\Log\AbstractLogger
      * @var      Handler\HandlerInterface Handler Interface
      * @param    LogInfo $logInfo
      */
-    public function __construct(array $handlers = array(), Info\InfoInterface $logInfo = null)
+    public function __construct(array $handlers = array(), Info\InfoInterface $logInfo = null, $autoRemove = false)
     {
         $this->logInfo = (isset($logInfo)) ? $logInfo : new Info\GenericInfo();
+        $this->autoRemove = $autoRemove;
 
         foreach ($handlers as $handlerName => $handlerObject) {
+            if (empty($handlerName)) {
```

Comment:
if you instantiate a logger like this:
```php
new logger([new handler1, new handler2, new handler3]
```

you will end up with an indexed array of
```php
0 => handler1
1 => handler2
2 => handler3
```

and on the first handler the name will return true for 

```php
empty($handlername)
```

perhaps for empty && !numeric ?

Reply:
so actually it can never happen that a handler has no name, since they are stores in an array